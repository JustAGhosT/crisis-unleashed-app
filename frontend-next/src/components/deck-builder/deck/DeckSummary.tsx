import React from "react";

interface DeckSummaryProps {
  uniqueCount: number;
  totalCount: number;
  averageCost: string;
}

export const DeckSummary: React.FC<DeckSummaryProps> = ({
  uniqueCount,
  totalCount,
  averageCost,
}) => (
  <div className="mt-4 pt-4 border-t border-slate-600">
    <div className="grid grid-cols-3 gap-4 text-center">
      <div>
        <div className="text-lg font-semibold text-purple-400">
          {uniqueCount}
        </div>
        <div className="text-xs text-gray-400">Unique Cards</div>
      </div>
      <div>
        <div className="text-lg font-semibold text-blue-400">{totalCount}</div>
        <div className="text-xs text-gray-400">Total Cards</div>
      </div>
      <div>
        <div className="text-lg font-semibold text-green-400">
          {averageCost}
        </div>
        <div className="text-xs text-gray-400">Avg Cost</div>
      </div>
    </div>
  </div>
);

DeckSummary.displayName = "DeckSummary";
