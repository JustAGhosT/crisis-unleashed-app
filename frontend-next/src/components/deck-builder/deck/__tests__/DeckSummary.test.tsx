import React from "react";
import { render } from "@testing-library/react";
import { DeckSummary } from "../DeckSummary";

describe("DeckSummary", () => {
  it("renders counts and average", () => {
    const { container, getByText } = render(
      <DeckSummary uniqueCount={10} totalCount={30} averageCost={"2.7"} />,
    );
    expect(getByText("Unique Cards")).toBeInTheDocument();
    expect(getByText("Total Cards")).toBeInTheDocument();
    expect(getByText("Avg Cost")).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
