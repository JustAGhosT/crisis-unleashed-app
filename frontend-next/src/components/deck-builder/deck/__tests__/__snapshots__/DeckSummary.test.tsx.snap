// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckSummary renders counts and average 1`] = `
<div
  class="mt-4 pt-4 border-t border-slate-600"
>
  <div
    class="grid grid-cols-3 gap-4 text-center"
  >
    <div>
      <div
        class="text-lg font-semibold text-purple-400"
      >
        10
      </div>
      <div
        class="text-xs text-gray-400"
      >
        Unique Cards
      </div>
    </div>
    <div>
      <div
        class="text-lg font-semibold text-blue-400"
      >
        30
      </div>
      <div
        class="text-xs text-gray-400"
      >
        Total Cards
      </div>
    </div>
    <div>
      <div
        class="text-lg font-semibold text-green-400"
      >
        2.7
      </div>
      <div
        class="text-xs text-gray-400"
      >
        Avg Cost
      </div>
    </div>
  </div>
</div>
`;
